test/statefulset_storage_test.go

```diff
@@ -0,0 +1,490 @@
+						sts.Spec.Template.Spec.InitContainers[0].VolumeMounts = initcontainerMountedVolumes
+						sts.Spec.Template.Spec.Volumes = volumes
+
+						err = updateSTS(k8sClient, ctx, sts)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						sts, err = getSTSFromRackID(aeroCluster, 0)
+						Expect(err).ToNot(HaveOccurred())
+
+						rackPodList, err := getRackPodList(k8sClient, ctx, sts)
+						Expect(err).ToNot(HaveOccurred())
+
+						// restart pod
+						for _, pod := range rackPodList.Items {
+							err := k8sClient.Delete(ctx, &pod)
+							Expect(err).ToNot(HaveOccurred())
+							started := waitForPod(&pod)
+							Expect(started).To(
+								BeTrue(), "pod was not able to come online in time",
+							)
+						}
+
+						isPresent, err := validateExternalVolumeInContainer(sts, 0)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(isPresent).To(
+							BeTrue(), "Unable to find volume",
+						)
+
+						isPresent, err = validateExternalVolumeInContainer(sts, 1)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(isPresent).To(
+							BeTrue(), "Unable to find volume",
+						)
+
+						isPresent, err = validateExternalVolumeInInitContainer(sts, 0)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(isPresent).To(
+							BeTrue(), "Unable to find volume",
+						)
+						aeroCluster, err = getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+
+						volume := asdbv1beta1.VolumeSpec{
+							Name: "newvolume",
+							Source: asdbv1beta1.VolumeSource{
+								EmptyDir: &v1.EmptyDirVolumeSource{},
+							},
+							Aerospike: &asdbv1beta1.AerospikeServerVolumeAttachment{
+								Path: "/newvolume",
+							},
+						}
+
+						aeroCluster.Spec.Storage.Volumes = append(
+							aeroCluster.Spec.Storage.Volumes, volume,
+						)
+						err = updateAndWait(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						isPresent, err = validateExternalVolumeInContainer(sts, 0)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(isPresent).To(
+							BeTrue(), "Unable to find volume",
+						)
+
+						isPresent, err = validateExternalVolumeInContainer(sts, 1)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(isPresent).To(
+							BeTrue(), "Unable to find volume",
+						)
+
+						isPresent, err = validateExternalVolumeInInitContainer(sts, 0)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(isPresent).To(
+							BeTrue(), "Unable to find volume",
+						)
+
+					},
+				)
+			},
+		)
+	},
+)
+
+func getSTSFromRackID(aeroCluster *asdbv1beta1.AerospikeCluster, rackID int) (
+	*appsv1.StatefulSet, error,
+) {
+	found := &appsv1.StatefulSet{}
+	err := k8sClient.Get(
+		goctx.TODO(),
+		getNamespacedNameForSTS(aeroCluster, rackID),
+		found,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return found, nil
+}
+
+func validateExternalVolumeInContainer(sts *appsv1.StatefulSet, index int) (
```

Comment:
validateExternalVolumeInContainer, validateExternalVolumeInInitContainer 
can be merged into a single function, take a flag as param to distinguish container or initContainer